### DISCUSSION TITLE
How are Servre Components rendered on subsequent navigations?

### DISCUSSION THREAD

**Discussion by ThisIsLoui** [vercel/next.js#65159] [Help] [answered]
### Summary

Hi!

I'm confused about how Server Components are rendered on subsequent navigations.

It's written in the docs that on subsequent navigations, React will use the RSC Payload to reconcile the Client and Server Component trees after downloading and parsing Client Component JavaScript bundle.

So, what if there are new server components to render on the new route? How will these new server components be rendered? Will the RSC Payload be refetched? 

I've searched the Internet and can't find an answer. Any help would be greatly appreciated.

Thank you in advance.

### Additional information

You can find the relevant content in the docs here: https://nextjs.org/docs/app/building-your-application/rendering/client-components#subsequent-navigations

**Comment by icyJoseph** [comment, marked as answer]
It is explained here I think, https://nextjs.org/docs/app/building-your-application/caching#static-and-dynamic-rendering:

<img width="611" alt="Screenshot 2024-04-29 at 11 04 56" src="https://github.com/vercel/next.js/assets/21013447/da7e46d3-3c6c-46ba-922d-eff44ccc6f30">

**Reply by ThisIsLoui** [reply]
Thanks for pointing this out. It really helps me a lot. 

So just to make it clear for anyone with the same question, it's written in the docs that on subsequent navigations or during prefetching:

>if the route segments are not in the cache, Next.js will fetch the React Server Components Payload from the server, and populate the Router Cache on the client.
